src/model_DAforcing.py

```diff
@@ -117,9 +118,472 @@ def __init__(self, bmi_cfg_file=None):
+                self.rfc_discharges = _rfc_discharges
+                self.rfc_timeseries_idx = _rfc_timeseries_idx
+                self.rfc_use_rfc = _rfc_use_rfc
+                self.rfc_Datetime = _rfc_Datetime
+                self.rfc_timeSteps = _rfc_timeSteps
+                self.rfc_StationId_array = _rfc_StationId_array
+                self.rfc_StationId_stringLengths = _rfc_StationId_stringLengths
+                self.rfc_List_array = _rfc_List_array
+                self.rfc_List_stringLengths = _rfc_List_stringLengths
+
+
+            testRevert = True
+            # The following code segments revert the array troute -> bmi-flattened
+            # array conversions, and are placed here only temporarily for
+            # verification purposes, to be moved to DataAssimilation eventually.
+            # If "reverse" code is not removed, it can be temporarily disabled
+            # by setting the testRevert flag to False
+            if (testRevert):
+
+                # USGS dataframe
+
+                # Unflatten the arrays
+                df_raw_usgs = _unflatten_array(self.usgsArray,self.nDates_usgs,\
+                                          self.nStations_usgs)
+
+                # Decode time/date axis
+                timeAxisName = 'time'
+                freqString = '5T'
+                df_withDates_usgs = _time_retrieve_from_arrays(df_raw_usgs, self.dateNull, \
+                            self.datesSecondsArray_usgs, timeAxisName, freqString)
+
+                # Decode station ID axis
+                stationAxisName = 'stationId'
+                df_withStationsAndDates_usgs = _stations_retrieve_from_arrays(df_withDates_usgs, \
+                    self.stationArray_usgs, self.stationStringLengthArray_usgs, \
+                    stationAxisName)
+
+                # Reservoir USGS dataframe
+
+                # Unflatten the arrays
+                df_raw_reservoirUsgs = _unflatten_array(self.reservoirUsgsArray,self.nDates_reservoir_usgs,\
+                                          self.nStations_reservoir_usgs)
+                
+                # Decode time/date axis
+                timeAxisName = 'time'
+                freqString = '15T'
+                df_withDates_reservoirUsgs = _time_retrieve_from_arrays(df_raw_reservoirUsgs, self.dateNull, \
+                            self.datesSecondsArray_reservoir_usgs, timeAxisName, freqString)
+
+                # Decode station ID axis
+                stationAxisName = 'stationId'
+                df_withStationsAndDates_reservoirUsgs = _stations_retrieve_from_arrays(df_withDates_reservoirUsgs, \
+                    self.stationArray_reservoir_usgs, self.stationStringLengthArray_reservoir_usgs, \
+                    stationAxisName)
+                
+                # Reservoir USACE dataframe
+
+                # Unflatten the arrays
+                df_raw_reservoirUsace = _unflatten_array(self.reservoirUsaceArray,self.nDates_reservoir_usace,\
```

Comment:
Got the following error when running with USACE persistence set to `False`:

`AttributeError: 'DAforcing_model' object has no attribute 'reservoirUsaceArray'`

Looks like you need to make sure you set some of these slots to empty (empty dataframe, False, None, etc.) values if the DA feature is turned off.